pghoard/basebackup.py

```diff
@@ -110,28 +115,36 @@ def compress_directly_to_a_file(self, proc, basebackup_path):
                 stderr=proc.stderr,
                 output_obj=output_obj,
                 compression_algorithm=compression_algorithm,
-                rsa_public_key=rsa_public_key)
+                rsa_public_key=rsa_public_key,
+                stderr_deque=stderr_deque)
```

Comment:
instead of passing in an explicit `stderr_deque` here I'd rather provide a callback which we can use to handle the output. compressor shouldn't need to care too much about the details of how we're going to use it.  If you don't care about the line splits the api could be as simple as:
```
stderr=proc.stderr,
stderr_sink=stderr_deque.append,
```